packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/models/auth_matrix_method_request.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:copy_with_extension/copy_with_extension.dart';
+import 'package:equatable/equatable.dart';
+import '../methods/otp/models/auth_matrix_otp_payload.dart';
+import '../methods/pin_biometric/models/auth_matrix_pin_code_payload.dart';
+import 'auth_matrix_method.dart';
+import 'payload/auth_matrix_payload_request.dart';
+
+part 'auth_matrix_method_request.g.dart';
+
+@CopyWith()
+class AuthMatrixMethodRequest with EquatableMixin {
+  AuthMatrixMethodRequest({
+    required this.securityToken,
+    required this.payload,
+  });
+
+  /// The security token used to authenticate the user and authorize the request.
+  final String securityToken;
+
+  /// The request body that contains the necessary user data to authenticate the user.
+  final AuthMatrixPayloadRequest payload;
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'securityToken': securityToken,
+        'payload': payload.toJson(),
+      };
+
+  factory AuthMatrixMethodRequest.fromJson(Map<String, dynamic> json) =>
```

Comment:
Could we use `@JsonSerializable` here as well?

Reply:
Due to the needed customisation the JsonSerializable annotation can not be used.